proctor-consumer/src/main/java/com/indeed/proctor/consumer/AbstractGroups.java

```diff
@@ -161,6 +173,9 @@ protected final Optional<TestBucket> getActiveBucket(final String testName) {
             if (testUsageObserver != null) {
                 testUsageObserver.markUsedForToggling(testName);
             }
+            if (exposureLogger != null) {
+                exposureLogger.createExposureLogEntry();
```

Comment:
The test observer is used only in a couple of cases and is used only in QA environment. If we use test observer, it means we have to call "testUsageObserver.markUsedForToggling(testName);" which add unnecessary overhead and risk to computation cost

Reply:
"we have to call "testUsageObserver.markUsedForToggling(testName);" which add unnecessary overhead and risk to computation cost" would you explain more why it will be more overhead and computation cost?

My bigger concerns will be just using a test observer itself will require users to do the instrumentation. That may be the reason why the usage is so low.

In order to auto instrument with observer, we may need to do things like changing the autogen code.

Reply:
I tried autogen code in proctor-interal, it can create an instance of ExposureLogger implementation class, so that user don't need to do any code change in their end